src/Microsoft.OpenApi/Models/OpenApiSchema.cs

```diff
@@ -566,6 +566,10 @@ internal void WriteAsItemsProperties(IOpenApiWriter writer)
             writer.WriteProperty(OpenApiConstants.Type, Type);
 
             // format
+            Format ??= AllOf?.FirstOrDefault(static x => x.Format != null)?.Format ??
+                    AnyOf?.FirstOrDefault(static x => x.Format != null)?.Format ??
+                    OneOf?.FirstOrDefault(static x => x.Format != null)?.Format;
```

Comment:
I'm not super familiar with the code base here. But shouldn't this have a condition to only do it when we are serializing to v2?

Also, the conditions should be string.IsNullOrEmpty, not just a null comparison.

Reply:
This method is only called by methods that serialize as v2